src/terraform/command_helpers.go

```diff
@@ -14,79 +14,79 @@ import (
 
 // CompileVarFlags compiles the variable flags passed into a Terraform command
 func CompileVarFlags(deployConfig deploy.Config, secrets types.Secrets, imagesMap map[string]string) ([]string, error) {
-	vars := compileSecrets(secrets)
-	imageVars := compileDockerImageVars(deployConfig, imagesMap)
-	vars = append(vars, imageVars...)
-	envVars, err := compileServiceEnvVars(deployConfig, secrets)
+	varFlagsMap := CompileDockerImageVars(deployConfig, imagesMap)
+	envVars, err := CompileServiceEnvVars(deployConfig, secrets)
 	if err != nil {
 		return []string{}, errors.Wrap(err, "cannot compile service environment variables")
 	}
-	vars = append(vars, envVars...)
-	dependencyVars, err := compileDependencyVars(deployConfig)
+	util.Merge(varFlagsMap, envVars)
+	dependencyVars, err := CompileDependencyVars(deployConfig)
 	if err != nil {
 		return []string{}, errors.Wrap(err, "cannot compile dependency variables")
 	}
-	vars = append(vars, dependencyVars...)
-	return append(vars, "-var", fmt.Sprintf("aws_profile=%s", deployConfig.AwsConfig.Profile)), nil
+	util.Merge(varFlagsMap, dependencyVars)
+	util.Merge(varFlagsMap, map[string]string{"aws_profile": deployConfig.AwsConfig.Profile})
+	util.Merge(varFlagsMap, secrets)
+	return BuildFlags(varFlagsMap), nil
```

Comment:
Thoughts on making the variable names:
* varFlagsMap -> varMap
* envVars -> servicesVarMap
* dependencyVars -> dependenciesVarMap

And the function names:

* CompileDockerImageVars -> GetDockerImageVarMap
* CompileServiceEnvVars -> GetServicesVarMap
* CompileDependencyVars -> GetDependenciesVarMap

Just want to make these a bit more consistent. Also within the functions use `varMap` as the return value